clippy_lints/src/doc.rs

```diff
@@ -258,6 +259,18 @@ fn check_word(cx: &EarlyContext, word: &str, span: Span) {
         s != "_" && !s.contains("\\_") && s.contains('_')
     }
 
+    if let Ok(url) = Url::parse(word) {
```

Comment:
We may want to have a configurable list of protocols, prepopulated with `["file", "http", "https", "ftp", "ssh", "mailto"]`.